build-system/git.js

```diff
@@ -26,7 +26,7 @@ const {getStdout} = require('./exec');
  * @return {string}
  */
 function gitMergeBaseMain() {
-  return getStdout(`git merge-base main HEAD`).trim();
+  return getStdout(`git merge-base origin/main HEAD`).trim();
```

Comment:
Let's add an explanatory comment here reflective of the PR description

Reply:
Went one step further and added logic to use `main` during local development and `origin/main` during CI. Also added an explanatory comment as suggested.